rebar.config

```
@@ -36,5 +38,8 @@
     verbose
 ]}.
 
-{pre_hooks, [{"", compile, "escript enc compile"}]}.
-{post_hooks, [{"", clean, "escript enc clean"}]}.
+{provider_hooks,
+ [{pre, [
+    {compile, {pc, compile}},
+    {clean, {pc, clean}}
+  ]}]}.
```

Don't know much about rebar and the pc plugin but shouldn't we keep the pre and post hooks?

```suggestion
{provider_hooks, [{pre, [{compile, {pc, compile}}]}
                  {post, [{compile, {pc, clean}}]}]}
```

---

Wow, well spotted!

---

Great find, thank you @jbernardo95 !

---

Wait! No. @jbernardo95  So your suggestion kinda works but not how it was before and not ideal either.

There are 2 hooks here: `compile` and `clean` and both are intended to run before the respective rebar3 commands. In theory the clean could be a post hook, but irrelevant.

So the intention is that `{pc, clean}` runs when someone issues `rebar3 clean`.

What your suggestion does is cleaning up immedately after `rebar3 compile`. Basically it will force `pc` to relink `jiffy.so` every time during compile, even though the artifact already exists.

---

As I said, I don't know how rebar works and what these configs mean exactly I just asked the question based on the diff x)

What you said makes sense to me and seems to [be on par with the documentation](https://github.com/blt/port_compiler#use-with-existing-dependency) 👍